enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -295,6 +288,133 @@ configures Envoy accordingly.  Configuring a load balancer and DNS enables
 clients to connect through Envoy to backend applications as specified by the
 Gateway and HTTPRoute objects.
 
+Following is the ServiceMeshControlPlane CR that the Ingress Operator creates:
+
+```yaml
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: gwapi
+  namespace: openshift-ingress
+spec:
+  profiles:
+  - gateway-controller
+  runtime:
+    components:
+      pilot:
+        container:
+          env:
+            PILOT_GATEWAY_API_CONTROLLER_NAME: openshift.io/gateway-controller
+  version: v2.4
+  techPreview:         # [1]
+    controlPlaneMode:
+      ClusterScoped
+```
+[1]: Specifying `techPreview.controlPlaneMode: ClusterScoped` is currently required even when specifying `spec.profiles: gateway-controller` due to a known issue in OSSM.
+
+This SMCP uses OSSM's built-in "gateway-controller" profile to enable Gateway
+API.  Specifying this profile is equivalent to specifying the following
+configuration on the SMCP:
+
+```yaml
+spec:
+  techPreview:
+    controlPlaneMode:
+      ClusterScoped
+    gatewayAPI:
+      enabled: true
+      controllerMode: true
+  addons:
+    grafana:
+      enabled: false
+    jaeger:
+      install: {}
+      name: jaeger
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  gateways:
+    egress:
+      enabled: false
+    ingress:
+      enabled: true
+      ingress: true
+  policy:
+    type: Istiod
+  security:
+    manageNetworkPolicy: false
+  tracing:
+    type: None
+```
```

Comment:
What about:
```
  runtime:
    components:
      pilot:
        container:
          env:
            PILOT_ENABLE_GATEWAY_API: "true"
            PILOT_ENABLE_GATEWAY_API_STATUS: "true"
            PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER: "true"
```
I think Daniel mention some of these were possibly going to be encapsulated in the spec.ingress.gateway, but I'm not so sure. Where did you pull this information from?

Reply:
Right, specifying `profiles: ["gateway-controller"]` should be equivalent to specifying `spec.techPreview.gatewayAPI.enabled` and `spec.techPreview.gatewayAPI.controllerMode` (as well as some other options), which should be equivalent to specifying the environment variables that you mentioned.  I have yet to test this; I'm basing this on Daniel's demo on 2023-01-16 and the follow-up discussion on 2023-02-02.

Reply:
Yea I remember Daniel mentioning that you couldn't actually enable Gateway API without enabling `PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER=true`, but I think he mentioned he was going to try to change that.